.github/workflows/build.yml

```diff
@@ -308,3 +310,195 @@ jobs:
+            });
+
+            console.log(`Created release ${release.data.tag_name} ${release.data.draft} ${release.data.prerelease} ${doIProceed}`);
+          }
+          console.log(`Release does exist with tag ${release.data.tag_name} [${release.data.draft} ${release.data.prerelease}]`);
+
+          // At this stage both tag & release exist
+
+          // Update nightly tag
+          await github.rest.git.updateRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: 'tags/nightly',
+            sha: ref_head.data.object.sha,
+          });
+          console.log(`Updated tag with sha ${ref_nightly.data.object.sha}`);
+
+          // Update the release
+          await github.rest.repos.updateRelease({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            release_id: release.data.id,
+            tag_name: 'nightly',
+            name: 'nightly',
+            body: 'Nightly release crafted with ♥️ somewhere on 🌎',
+            draft: false,
+            prerelease: true,
+          });
+          console.log(`Updated ${release.data.tag_name} nightly release  ${release.data.draft} ${release.data.prerelease}`);
+
+          // Get all tags and keep the oldest one starting by v
+          let oldest_tag = { name: 'v0.0.0' };
+
+          try {
+            const tags = await github.rest.repos.listTags({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+
+            // Keep latest tag
+            for (const tag of tags.data) {
+              if (tag.name.startsWith('v')) {
+                if (tag.name > oldest_tag.name) {
+                  oldest_tag = tag;
+                }
+              }
+            }
+          } catch (err) {
+            console.log(err);
+          }
+          console.log(`Previous release has tag ${oldest_tag.name} → ${oldest_tag.commit.sha}`);
+
+          // Get current date with format %Y-%m-%d
+          const today = new Date();
+          var ISOToday = today.toISOString().split('T')[0];
+          console.log(`Today: ${ISOToday}`);
+
+          // Count all commits between HEAD and oldest tag
+          let distance = 0;
+          try {
+            let page = 1;
+            let found = false;
+            do {
+              const response = await github.rest.repos.listCommits({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                per_page: 100,
+                page: page,
+              });
+              response.data.forEach(commit => {
+                if (commit.sha === oldest_tag.commit.sha) {
+                  found = true;
+                } else {
+                  if (found !== true) distance++;
+                }
+              });
+              page++;
+            } while (found === false && response.length > 0);
+            console.log(`Found ${distance} commits`);
+          } catch (err) {
```

Comment:
if something went wrong we likely want to return or throw

Reply:
fixed

Reply:
I found a simpler solution:
```js
       // Limited to 250 commits
       const distance = await github.rest.repos.compareCommitsWithBasehead({
         owner: context.repo.owner,
         repo: context.repo.repo,
         basehead: `${oldest_tag.commit.sha}...${master_sha}`,
       }).then(d => d.data.total_commits);
```
If there are more than 250 commits, the list of nighly wheels is not manageable and I am not sure that having wheels with the same distance won't make any significant difference.

Reply:
Indeed. I learned something